src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotNodeSummary.java

```diff
@@ -48,11 +48,20 @@
   private static final Logger LOG = LogManager.getLogger(HotNodeSummary.class);
   private final String nodeID;
   private final String hostAddress;
+  private final List<HotShardSummary> hotShardSummaryList;
```

Comment:
can we remove this private list and add all HotShardSummary into its GenericSumary list ? The idea is HotNodeSummry is a generic type of summary which does not need to know what type of summary it has in its nested summary list.

Reply:
As discussed, using the generic list would mean we use the `instanceof` to decide the list type, which is something we want to avoid.